react/components/admin/redirects/Form/index.tsx

```diff
@@ -158,54 +156,39 @@ class Form extends Component<Props, State> {
               <FormFieldSeparator />
               {shouldShowDatePicker && (
                 <Fragment>
-                  {this.isViewMode ? (
-                    <Input
-                      disabled
-                      label={intl.formatMessage({
-                        id: 'pages.admin.redirects.form.datePicker.title',
-                      })}
-                      value={getFormattedLocalizedDate(data.endDate, locale)}
+                  <FormattedMessage id="pages.admin.redirects.form.datePicker.title">
+                    {text => <div className="mb3 w-100 f6">{text}</div>}
+                  </FormattedMessage>
+                  <div className="flex">
+                    <DatePicker
+                      direction="up"
+                      locale={locale}
+                      onChange={this.updateEndDate}
+                      useTime={true}
+                      minDate={new Date()}
```

Comment:
I think it'd be better to instantiate this date once in the constructor and store it in a private attribute, and then reference it here.